### PR TITLE
Add pallet-logion-vote to node-api

### PR DESCRIPTION
**Pull request by benoitdevos**
* Add pallet-logion-vote to node-api.
* Migrate from CommonJS to ESM. 

logion-network/logion-internal#713

### CODE REVIEW

File: packages/client/tsconfig.test.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    "module": "es2020",
+    "target": "es2020",
```
Comment (gdethier) [inline packages/client/tsconfig.test.json:5]: This is redundant with the configuration in extended config.

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "website"
   ],
   "packageManager": "yarn@3.2.0",
+  "type": "module",
```
Comment (gdethier) [inline package.json:15]: Is this required?

File: packages/node-api/src/interfaces/augment-api-events.ts [hunk 3]
```diff
@@ -392,5 +404,15 @@ declare module '@polkadot/api-base/types/events' {
        **/
       [key: string]: AugmentedEvent<ApiType>;
     };
+    vote: {
+      /**
+       * Issued upon new Vote creation. [voteId, legalOfficers]
+       **/
+      VoteCreated: AugmentedEvent<ApiType, [u64, Vec<AccountId32>]>;
```
Comment (gdethier) [inline packages/node-api/src/interfaces/augment-api-events.ts:411]: So the list of voters is still part of the event after all? Currently not a big deal, but may become a problem in the future as the number of voters increases.

File: packages/node-api/test/ChainTime.test.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { ApiPromise } from "@polkadot/api";
 import { CURRENT_BLOCK_NUMBER } from "./__mocks__/PolkadotApiMock";
-import { ChainTime } from "../src/ChainTime";
+import { ChainTime } from "../src";
```
Comment (gdethier) [inline packages/node-api/test/ChainTime.test.ts:3]: Strange that this works without explicitly pointing to the `index.js`.

File: packages/node-api/test/Recovery.test.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { DEFAULT_LEGAL_OFFICER } from "./TestData";
 import { ApiPromise } from '@polkadot/api';
 import { setQueryRecoveryRecoverable, setQueryRecoveryActiveRecoveries } from './__mocks__/PolkadotApiMock';
-import { createRecovery, getRecoveryConfig, initiateRecovery, getActiveRecovery } from '../src/Recovery';
+import { createRecovery, getRecoveryConfig, initiateRecovery, getActiveRecovery } from '../src';
```
Comment (gdethier) [inline packages/node-api/test/Recovery.test.ts:4]: Strange that this works without explicitly pointing to the `index.js`.

File: packages/node-api/test/Vault.test.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { getVaultAddress } from "../src/Vault";
+import { getVaultAddress } from "../src";
```
Comment (gdethier) [inline packages/node-api/test/Vault.test.ts:1]: Strange that this works without explicitly pointing to the `index.js`.

File: packages/node-api/test/numbers.test.ts [hunk 7]
```diff
@@ -9,7 +9,7 @@ import {
     ScientificNumber,
     MEGA,
     ATTO, PICO, convertToPrefixed,
-} from '../src/numbers';
+} from '../src';
```
Comment (gdethier) [inline packages/node-api/test/numbers.test.ts:12]: Strange that this works without explicitly pointing to the `index.js`.

File: packages/client/tsconfig.json [hunk 8]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "module": "es2020",
+    "target": "es2020",
```
Comment (gdethier) [inline packages/client/tsconfig.json:5]: Redundant with the configuration of extended config.

File: packages/node-api/tsconfig.json [hunk 9]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "module": "es2020",
+    "target": "es2020",
```
Comment (gdethier) [inline packages/node-api/tsconfig.json:5]: Redundant with extended config.

File: packages/node-api/tsconfig.test.json [hunk 10]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    "module": "es2020",
+    "target": "es2020",
```
Comment (gdethier) [inline packages/node-api/tsconfig.test.json:5]: Redundant with extended config.

**Review by gdethier — APPROVED** [review]
I have a few remarks and a couple of questions but nothing critical.

File: packages/client/tsconfig.test.json (hunk 1 above)
Comment (benoitdevos) [reply packages/client/tsconfig.test.json]: Removed.

File: package.json (hunk 2 above)
Comment (benoitdevos) [reply package.json]: No. Removed.

File: packages/node-api/tsconfig.json (hunk 9 above)
Comment (benoitdevos) [reply packages/node-api/tsconfig.json]: Removed.

File: packages/client/tsconfig.json (hunk 8 above)
Comment (benoitdevos) [reply packages/client/tsconfig.json]: Removed.

File: packages/node-api/tsconfig.test.json (hunk 10 above)
Comment (benoitdevos) [reply packages/node-api/tsconfig.test.json]: Removed.

File: packages/node-api/src/interfaces/augment-api-events.ts (hunk 3 above)
Comment (benoitdevos) [reply packages/node-api/src/interfaces/augment-api-events.ts:411]: pallet et node-api will be updated at the node runtime.
Reminder:
https://github.com/logion-network/logion-internal/issues/718

**Pull request merged**